test/unit/features/debug.spec.js

```diff
@@ -80,4 +80,38 @@ found in
            <Root>`
     ).toHaveBeenWarned()
   })
+
+  describe('warn', () => {
+    const msg = 'message'
+    const vm = new Vue()
+
+    it('calls warnHandler if warnHandler is set', () => {
+      Vue.config.warnHandler = jasmine.createSpy()
+
+      warn(msg, vm)
+
+      expect(Vue.config.warnHandler).toHaveBeenCalledWith(msg, vm, jasmine.any(String))
+
+      Vue.config.warnHandler = null
+    })
+
+    it('calls console.error if silent is false', () => {
+      Vue.config.silent = false
+
+      warn(msg, vm)
+
+      expect(msg).toHaveBeenWarned()
```

Comment:
`toHaveBeenWarned` currently still regards something being "warned" when it calls `console.error` (`warnHandler` is not set). Did we want to change this so it also tracks if a custom warnHandler has been called?